src/app/core/services/local-storage.service.ts

```diff
@@ -0,0 +1,23 @@
+import {Injectable} from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LocalStorageService<T> {
```

Comment:
Nit, but this service does both contain an API to affect the entire store and has methods to work with individual keys, yet is hard typed on a single type.

we could either:
- lock the unit down to a single key + type, remove the clearData method, then it's similar to the "local stored value" and could be merged with it
- make this a generic service, move the type templating to the saveData and getData methods, e.g. `saveData<T>(key: string, value: T)`.